lnwire/query_short_chan_ids.go

```diff
@@ -152,6 +152,10 @@ func decodeShortChanIDs(r io.Reader) (ShortChanIDEncoding, []ShortChannelID, err
 	// type was specified, meaning that there're no further bytes to be
 	// parsed.
 	if len(queryBody) == 0 {
+		// Check whether the encodingType is valid or not.
+		if encodingType > EncodingSortedZlib {
+			return 0, nil, ErrUnknownShortChanIDEncoding(encodingType)
+		}
```

Comment:
is it possible just to remove this whole check? to me it looks like both of the existing sections below handle the zero-length case gracefully, and then error will instead be produced by the `default` case

Reply:
Yeah I can just remove this check

Reply:
Removed the check, but had to add a special check [here](https://github.com/lightningnetwork/lnd/pull/1895/files#diff-172a0738c715b4304ad9bbc83ed046c4R90). The first `ReadMessage` call would succeed and return an empty list of `ShortChannelIDs` (it would succeed because there would be bytes, but not enough to be a zlib-encoded `*ShortChannelID`); the subsequent call would error on `ReadMessage` when creating the zlib reader because there were no more bytes left to be read.

Reply:
On second thought, removing this check makes `TestReplyChannelRangeEmpty` fail. Now any `QueryShortChanIDs` or `ReplyChannelRange` message with zlib encoding and no `ShortChannelIDs` will fail with `unable to create zlib reader: unexpected EOF`.  I don't think the peer should fail on a legitimate message like this.